### PR TITLE
feat: Ask Users to Invite Team Members During Onboarding

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

Adds an onboarding step for inviting team members and a feature flag to toggle whether this step is shown. I'm going to run an experiment with this flag to see if activation rates for product analytics are affected. 

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/onboarding/OnboardingInviteTeammates.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    const { invitesToSend, invites } = useValues(inviteLogic)
+    const { appendInviteRow, deleteInvite, updateMessage } = useActions(inviteLogic)
+
+    const invitesReversed = invites.slice().reverse()
+    const areInvitesCreatable = invitesToSend.length + 1 < MAX_INVITES_AT_ONCE
+    const areInvitesDeletable = invitesToSend.length > 1
+
+    return (
+        <OnboardingStep title={`Invite teammates to ${user?.organization?.name || 'PostHog'}`} stepKey={stepKey}>
+            {preflight?.email_service_available ? (
```
Comment (xrdt) [inline frontend/src/scenes/onboarding/OnboardingInviteTeammates.tsx:27]: This is mainly copied from the `InviteModal.tsx`. I should probably extract more things into subcomponents / constant so it's easier to keep these modules in sync.

**Comment by xrdt** [comment]
Need to add feature flag to app.posthog.com

File: frontend/src/scenes/onboarding/Onboarding.tsx [hunk 2]
```diff
@@ -63,6 +65,10 @@ const OnboardingWrapper = ({ children }: { children: React.ReactNode }): JSX.Ele
             const OtherProductsStep = <OnboardingOtherProductsStep stepKey={OnboardingStepKey.OTHER_PRODUCTS} />
             steps = [...steps, OtherProductsStep]
         }
+        if (featureFlags[FEATURE_FLAGS.INVITE_TEAM_MEMBER_ONBOARDING]) {
```
Comment (raquelmsmith) [inline frontend/src/scenes/onboarding/Onboarding.tsx:68]: we need this flag to evaluate to `test`, here it's just looking for a truthy value

**Comment by xrdt** [comment]
New flow looks like this: 

https://github.com/PostHog/posthog/assets/21014901/d6b9eb25-0cfb-48a2-82ae-bd3f6fd2ef52



It should roll out to everybody, with 50% of people seeing the invite step, 50% not seeing it. 
New feature flag: https://app.posthog.com/feature_flags/23836
New experiment: https://app.posthog.com/experiments/8176

File: frontend/src/scenes/onboarding/OnboardingInviteTeammates.tsx (hunk 1 above)
Comment (raquelmsmith) [reply frontend/src/scenes/onboarding/OnboardingInviteTeammates.tsx:27]: Yeah it seems like it'd be good to extract things so it's not a bunch of duplicate code.

**Comment by xrdt** [comment]
Attempting to make the invite copy more interesting. 
Screenshot 2023-12-07 at 3 27 51 PM

File: frontend/src/scenes/onboarding/OnboardingInviteTeammates.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    return (
+        <OnboardingStep title={`PostHog is better with teammates`} stepKey={stepKey}>
+            <div className="flex items-center py-8 gap-8 mx-16">
+                <span>
+                    Invite your teammates so you can {productAnalyticsOnboarded && 'share your dashboard insights'}
+                    {surveysOnboarded &&
+                        `${productAnalyticsOnboarded ? ' and ' : ''}review your survey results together`}
+                    {sessionReplayOnboarded &&
+                        `${
+                            surveysOnboarded || productAnalyticsOnboarded ? 'and' : ''
```
Comment (xrdt) [inline frontend/src/scenes/onboarding/OnboardingInviteTeammates.tsx:28]: Not sure what I should add for feature flags...

**Pull request merged**